src/armory/Armory.js

```diff
@@ -65,158 +67,197 @@ class Armory extends React.Component<Props, State> {
+							this.setState({selectedWeaponOne: jsonObjectOfTanks[tank].components[1]});
+							this.setState({selectedWeaponTwo: jsonObjectOfTanks[tank].components[2]});
+							this.setState({selectedScannerOne: jsonObjectOfTanks[tank].components[3]});
+							this.setState({selectedScannerTwo: jsonObjectOfTanks[tank].components[4]});
+							this.setState({selectedScannerThree: jsonObjectOfTanks[tank].components[5]});
+							this.setState({selectedJammer: jsonObjectOfTanks[tank].components[6]});
+							this.setState({selectedThreads: jsonObjectOfTanks[tank].components[7]});
+							this.setState({selectedSingleUseItemOne: jsonObjectOfTanks[tank].components[8]});
+							this.setState({selectedSingleUseItemTwo: jsonObjectOfTanks[tank].components[9]});
+							this.setState({selectedSingleUseItemThree: jsonObjectOfTanks[tank].components[10]});
+							setTankForCasus(this.state.selectedTankId);
+						}
+						this.state.tankOptions.push(obj);
+					}
+					this.getUserInventory();
+				}
+			})
+		).catch(
+			(error) => {
+				console.log('Couldnt connect to server!');
+				console.log(error);
+				return error;
 			}
-			tankOptions.push(obj);
-		}
-		this.getUserInventory();
+		);
 	};
+
 	//This will get all the inventory from a user and fill out the arrays used in the front end for the backend
-	getUserInventory = async ():Promise<void> => {
-		const cookies = new Cookies();
-		const token = cookies.get('token');
-		const response = await fetch('/api/user/getUser/', {
-			method: 'GET',
-			headers: {
-				'Access-Control-Allow-Origin': '*',
-				'Content-Type': 'application/json',
-				'Access-Control-Allow-Credentials': 'true',
-				'x-auth-token': token
-			},
-		});
-		const jsonObjectOfUser = await response.json();
-		//set the users id
-		this.setState({userId:jsonObjectOfUser._id});
-		for (const component in jsonObjectOfUser.inventory.tankComponents) {
-			const typeOfItem = getTankComponent(verifyComponent(component));
-			//This will add the chassis that the user has
-			if(typeOfItem === 'chassis' && jsonObjectOfUser.inventory.tankComponents[component] > 0) {
-				let obj = {};
-				obj['value'] = component;
-				obj['label'] = component;
-				chassisOptions.push(obj);
+	getUserInventory () {
+		const responsePromise = getUser();
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+					return data;
+				}
+				else {
+					let jsonObjectOfUser = data;
+					for (const component in jsonObjectOfUser.inventory.tankComponents) {
+						const typeOfItem = getTankComponent(verifyComponent(component));
+						//This will add the chassis that the user has
+						if(typeOfItem === 'chassis' && jsonObjectOfUser.inventory.tankComponents[component] > 0) {
+							let obj = {};
+							obj['value'] = component;
+							obj['label'] = component;
```

Comment:
yikes, this has all sorts of issues.
1) obj, value, and label are very undescriptive names for people who don't know what this code does.
2) You copied and pasted `jsonObjectOfUser.inventory.tankComponents[component] > 0` six times instead of just filtering out the ones that were zero ahead of time
3) all sorts of lets that should be consts